### PR TITLE
Add myself as a code owner of the contributing documentation

### PR DESCRIPTION
**Pull request by infinisil**
I want to take ownership of this part of the documentation, especially with the cleanup in https://github.com/NixOS/nixpkgs/pull/245243. Doing so before that PR is merged also allows me to get notified about any potential future merge conflicts before they happen.

### CODE REVIEW

**Review by Janik-Haag — APPROVED** [review]
Sounds like a good idea.

**Comment by infinisil** [comment]
No need to wait for CI, and GitHub confirmed to me that

>  This CODEOWNERS file is valid.

File: .github/CODEOWNERS
Comment (infinisil) [reply .github/CODEOWNERS:63]: Oh damn yeah, sorry I merged it already, but I'll open a new one for that!

Comment (Janik-Haag) [reply .github/CODEOWNERS:63]: Argh sorry, I should have been more careful. Also why does github say the file is valid with a basically empty entry?

Comment (infinisil) [reply .github/CODEOWNERS:63]: I guess it's not technically invalid, it just doesn't make much sense, I wouldn't mind them adding a check for that. It does have a slight effect though because codeowners of subdirectories override code owners of the parents.

Comment (Lord-Valen) [reply .github/CODEOWNERS:63]: Because it actually is valid: A line with no owner is used to exclude a path from ownership. 
```
# In this example, @octocat owns any file in the `/apps`
# directory in the root of your repository except for the `/apps/github`
# subdirectory, as its owners are left empty.
/apps/ @octocat
/apps/github
```
- [Codeowners docs](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file)

Comment (infinisil) [reply .github/CODEOWNERS:63]: Fixed in https://github.com/NixOS/nixpkgs/pull/248168

**Pull request merged**
